src/coreclr/jit/flowgraph.cpp

```diff
@@ -754,29 +754,6 @@ GenTreeLclVar* Compiler::fgIsIndirOfAddrOfLocal(GenTree* tree)
     {
         GenTree* addr = tree->AsIndir()->Addr();
 
-        // Post rationalization, we can have Indir(Lea(..) trees. Therefore to recognize
-        // Indir of addr of a local, skip over Lea in Indir(Lea(base, index, scale, offset))
-        // to get to base variable.
-        if (addr->OperGet() == GT_LEA)
-        {
-            // We use this method in backward dataflow after liveness computation - fgInterBlockLocalVarLiveness().
-            // Therefore it is critical that we don't miss 'uses' of any local.  It may seem this method overlooks
-            // if the index part of the LEA has indir( someAddrOperator ( lclVar ) ) to search for a use but it's
-            // covered by the fact we're traversing the expression in execution order and we also visit the index.
-            GenTreeAddrMode* lea  = addr->AsAddrMode();
-            GenTree*         base = lea->Base();
-
-            if (base != nullptr)
-            {
-                if (base->OperGet() == GT_IND)
-                {
-                    return fgIsIndirOfAddrOfLocal(base);
-                }
-                // else use base as addr
-                addr = base;
-            }
-        }
-
```

Comment:
Couldn't we still have this (maybe not for tracked locals)?

Reply:
Right, I mean for untracked locals. The name of the function (`fgIsIndirOfAddrOfLocal`) does not seem to suggest that this function should only work for tracked locals, but I guess we don't really use it for any other purpose.

Reply:
Can we unify some of these functions now? `fgIsIndirOfAddrOfLocal`, `GenTree::IsLocalAddrExpr`, `GenTree::DefinesLocalAddr`, `impIsAddressInLocal`? I guess the latter one not, but maybe some of the others? (Not necessarily in this PR)